src/nccl_ofi_freelist.c

```diff
@@ -64,6 +65,7 @@ static int freelist_init_internal(size_t entry_size,
 
 	freelist->entry_size = NCCL_OFI_ROUND_UP(NCCL_OFI_MAX(entry_size, sizeof(struct nccl_ofi_freelist_elem_t)),
 						 NCCL_OFI_MAX(8, MEMCHECK_GRANULARITY));
+	freelist->entry_size = NCCL_OFI_ROUND_UP(freelist->entry_size, entry_alignment);
```

Comment:
You need to make sure that entry_size is 8 and MEMCHECK_GRANULARITY byte aligned.

Reply:
Oops, you're right. I think I need to use the MAX of 8, `MEMCHECK_GRANULARITY`, and `entry_alignment`.